src/libsync/owncloudpropagator.cpp

```diff
@@ -388,6 +395,9 @@ void OwncloudPropagator::start(const SyncFileItemVector& items)
                 currentDirJob->append(dir);
             }
             directories.push(qMakePair(item->destination() + "/" , dir));
+        } else if (enableScheduledRequests
+              && (item->_instruction == CSYNC_INSTRUCTION_NEW || item->_instruction == CSYNC_INSTRUCTION_SYNC)) {
+            filesJob->append(item);
```

Comment:
This changes one purpose of the PropagateDirectory structure: update the directory's etag in the database only once child files have been synced properly.

I can't wrap my head about the conditions where this could be an issue, but @ogoffart should know if this could cause concrete problems.

See e.g. https://github.com/owncloud/client/blob/1cec2ca13dc35aba6aaf79f42bde9a6dd00143a7/src/libsync/owncloudpropagator.cpp#L705